services/app-api/src/postgres/contractAndRates/submitContract.ts

```diff
@@ -96,6 +94,22 @@ async function submitContract(
                 },
             })
 
+            // we only want to update the rateRevision's submit info if it has not already been submitted
+            for (const rev of unsubmittedRates) {
+                await tx.rateRevisionTable.update({
```

Comment:
you could probably do an updateMany here or something but this is fine we aren't getting submissions with 100s of rates

Reply:
yeah that's a good point, I'll try to update that to an `updateMany`.